src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java

```diff
@@ -225,13 +228,15 @@ private void writeFileError(AuditEvent event) {
     @Override
     public void addException(AuditEvent event, Throwable throwable) {
         final String fileName = event.getFileName();
-        if (fileName == null || !fileMessages.containsKey(fileName)) {
-            writeException(throwable);
-        }
-        else {
+        final boolean fileNameExists = fileName != null && fileMessages.containsKey(fileName);
+
+        if (fileNameExists) {
```

Comment:
```suggestion
        if (fileMessages.containsKey(fileName)) {
```
Map#containsKey should be able to handle null